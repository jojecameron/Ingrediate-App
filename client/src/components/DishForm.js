import * as React from 'react';
import { useState, useEffect } from 'react';

const DishForm = (props) => {

  const [selectedDish, setSelectedDish] = useState('breakfast');

  useEffect(() => {
    props.handleDishChange(selectedDish);
  }, [selectedDish]);

  const handleDishChange = (event) => {
    setSelectedDish(event.target.name);
  };

  return (
    <form id='checkboxes'>
      <input 
        type='checkbox' 
        name='breakfast'
        checked={selectedDish === 'breakfast'}
        onChange={handleDishChange}
        />
      <label>Breakfast</label>
      <input 
        type='checkbox' 
        name='lunch'
        checked={selectedDish === 'lunch'}
        onChange={handleDishChange}
      />
      <label>Lunch</label>
      <input 
        type='checkbox' 
        name='dinner'
        checked={selectedDish === 'dinner'}
        onChange={handleDishChange}
        />
      <label>Dinner</label>
      <input 
        type='checkbox' 
        name='dessert'
        checked={selectedDish === 'dessert'}
        onChange={handleDishChange}
        />
      <label>Dessert</label>
    </form>
  );
}
  

export default DishForm;