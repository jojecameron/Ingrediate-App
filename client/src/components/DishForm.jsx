import * as React from "react";
import { useState, useEffect } from "react";

const DishForm = ({ setDishType, dishType }) => {

  const handleDishChange = (event) => {
    setDishType(event.target.name);
  };

  return (
    <form id="checkboxes">
      <div>
        <input
          type="checkbox"
          className="checkbox"
          name="breakfast"
          checked={dishType === "breakfast"}
          onChange={handleDishChange}
        />
        <label>Breakfast</label>
      </div>
      <div>
        <input
          type="checkbox"
          className="checkbox"
          name="lunch"
          checked={dishType === "lunch"}
          onChange={handleDishChange}        
        />
        <label>Lunch</label>
      </div>
      <div>
        <input
          type="checkbox"
          className="checkbox"
          name="dinner"
          checked={dishType === "dinner"}
          onChange={handleDishChange}       
        />
        <label>Dinner</label>
      </div>
      <div>
        <input
          type="checkbox"
          className="checkbox"
          name="dessert"
          checked={dishType === "dessert"}
          onChange={handleDishChange}        
        />
        <label>Dessert</label>
      </div>
    </form>
  );
};

export default DishForm;
