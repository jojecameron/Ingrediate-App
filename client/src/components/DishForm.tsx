import { DishFormProps, DishType } from "../types";

const DishForm = (props: DishFormProps): JSX.Element => {

  const { setDishType, dishType } = props;

  return (
    <form id="checkboxes">
      <div>
        <input
          type="radio"
          className="checkbox"
          name="breakfast"
          value="breakfast"
          checked={dishType === "breakfast"}
          onChange={(event) => setDishType(event.target.name as DishType)}
        />
        <label>Breakfast</label>
      </div>
      <div>
        <input
          type="radio"
          className="checkbox"
          name="lunch"
          value="lunch"
          checked={dishType === "lunch"}
          onChange={(event) => setDishType(event.target.name as DishType)}        
        />
        <label>Lunch</label>
      </div>
      <div>
        <input
          type="radio"
          className="checkbox"
          name="dinner"
          value="dinner"
          checked={dishType === "dinner"}
          onChange={(event) => setDishType(event.target.name as DishType)}       
        />
        <label>Dinner</label>
      </div>
      <div>
        <input
          type="radio"
          className="checkbox"
          name="dessert"
          value="dessert"
          checked={dishType === "dessert"}
          onChange={(event) => setDishType(event.target.name as DishType)}        
        />
        <label>Dessert</label>
      </div>
    </form>
  );
};

export default DishForm;
