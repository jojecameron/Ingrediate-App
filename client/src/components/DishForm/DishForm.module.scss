#checkboxes {
  align-items: center;
  color: $offWhite;
  display: flex;
  font-family: sans-serif;
  justify-content: space-evenly;
  padding: 10px;
  width: 40%;
}

@media screen and (max-width: 804px) {
  #checkboxes {
    align-items: flex-start;
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
  }
}

.checkbox {
  background-color: white;
  border: 2px solid #ddd;
  border-radius: 25%;
  cursor: pointer;
  height: 20px;
  margin-bottom: 6px;
  outline: none;
  vertical-align: middle;
  width: 20px;
}
