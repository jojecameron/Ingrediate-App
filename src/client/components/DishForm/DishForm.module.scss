#checkboxes {
  align-items: center;
  color: $offWhite;
  display: flex;
  font-family: sans-serif;
  justify-content: space-evenly;
  height: 62.1px;
}

@media screen and (max-width: 700px) {
  #checkboxes {
    align-items: flex-end;
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    height: auto;
  }
  .dish-form {
    align-self: flex-start;
  }
}

.checkbox {
  cursor: pointer;
  height: 20px;
  outline: none;
  vertical-align: middle;
  width: 20px;
  margin-top: 0px;
  margin-right: 5px;
}

.radio-label {
  margin: 0 10px;
  display: flex;
  justify-content: center;
  align-items: center;
  font-weight: 800;
}

.dish-form {
  align-self: baseline;
  display: flex;
}
