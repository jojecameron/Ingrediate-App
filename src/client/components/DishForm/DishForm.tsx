import { DishFormProps, DishType } from '../../types';

const DishForm = (props: DishFormProps): JSX.Element => {
  const { setDishType, dishType } = props;

  return (
    <div className="dish-form">
      <form id="checkboxes">
        <div className="radio-label">
          <label htmlFor="breakfast">Breakfast</label>
          <input
            type="radio"
            id="breakfast"
            className="checkbox"
            name="breakfast"
            value="breakfast"
            checked={dishType === 'breakfast'}
            onChange={(event) => setDishType(event.target.name as DishType)}
          />
        </div>
        <div className="radio-label">
          <label htmlFor="lunch">Lunch</label>
          <input
            type="radio"
            id="lunch"
            className="checkbox"
            name="lunch"
            value="lunch"
            checked={dishType === 'lunch'}
            onChange={(event) => setDishType(event.target.name as DishType)}
          />
        </div>
        <div className="radio-label">
          <label htmlFor="dinner">Dinner</label>
          <input
            type="radio"
            id="dinner"
            className="checkbox"
            name="dinner"
            value="dinner"
            checked={dishType === 'dinner'}
            onChange={(event) => setDishType(event.target.name as DishType)}
          />
        </div>
        <div className="radio-label">
          <label htmlFor="dessert">Dessert</label>
          <input
            type="radio"
            id="dessert"
            className="checkbox"
            name="dessert"
            value="dessert"
            checked={dishType === 'dessert'}
            onChange={(event) => setDishType(event.target.name as DishType)}
          />
        </div>
      </form>
    </div>
  );
};

export default DishForm;
