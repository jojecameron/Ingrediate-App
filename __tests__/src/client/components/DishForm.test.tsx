/**
 * `DishForm` is a component that allows the user to select a dish type from a list of options.
 * It takes two props:
 * - `dishType`: The currently selected dish type. Should be one of 'breakfast', 'lunch', 'dinner', or 'dessert'.
 * - `setDishType`: A function that's called when the user selects a new dish type.
 *
 * This test suite checks that the `DishForm` component works as expected.
 */

import React from 'react';
import { render } from '@testing-library/react';
import DishForm from '../../../../src/client/components/DishForm/DishForm';

function createMockProps(
  dishType: 'breakfast' | 'lunch' | 'dinner' | 'dessert',
) {
  return {
    setDishType: jest.fn(),
    dishType,
  };
}

describe('DishForm', () => {
  test('renders a basic snapshot', () => {
    const mockProps = createMockProps('breakfast');
    const { container } = render(<DishForm {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  const dishTypes = ['breakfast', 'lunch', 'dinner', 'dessert'] as const;

  dishTypes.forEach((dishType) => {
    test(`renders a snapshot for dishType: ${dishType}`, () => {
      const mockProps = createMockProps(dishType);
      const { container } = render(<DishForm {...mockProps} />);
      expect(container).toMatchSnapshot();
    });
  });

  test('has initial dishType state set to breakfast', () => {
    const mockProps = createMockProps('breakfast');
    const { getByLabelText } = render(<DishForm {...mockProps} />);
    const button = getByLabelText('Breakfast') as HTMLInputElement;
    expect(button.checked).toBe(true);
  });

  test('handles a dishType change to lunch', () => {
    const mockProps = createMockProps('breakfast');
    const { getByLabelText } = render(<DishForm {...mockProps} />);
    const button = getByLabelText('Lunch') as HTMLInputElement;
    button.click();
    expect(mockProps.setDishType).toHaveBeenCalledWith('lunch');
    expect(mockProps.setDishType).toHaveBeenCalledTimes(1);
  });

  test('handles a dishType change to dinner', () => {
    const mockProps = createMockProps('breakfast');
    const { getByLabelText } = render(<DishForm {...mockProps} />);
    const button = getByLabelText('Dinner') as HTMLInputElement;
    button.click();
    expect(mockProps.setDishType).toHaveBeenCalledWith('dinner');
    expect(mockProps.setDishType).toHaveBeenCalledTimes(1);
  });

  test('handles a dishType change to dessert', () => {
    const mockProps = createMockProps('breakfast');
    const { getByLabelText } = render(<DishForm {...mockProps} />);
    const button = getByLabelText('Dessert') as HTMLInputElement;
    button.click();
    expect(mockProps.setDishType).toHaveBeenCalledWith('dessert');
    expect(mockProps.setDishType).toHaveBeenCalledTimes(1);
  });

  test('handles a dishType change to breakfast', () => {
    const mockProps = createMockProps('dinner');
    const { getByLabelText } = render(<DishForm {...mockProps} />);
    const button = getByLabelText('Breakfast') as HTMLInputElement;
    button.click();
    expect(mockProps.setDishType).toHaveBeenCalledWith('breakfast');
    expect(mockProps.setDishType).toHaveBeenCalledTimes(1);
  });

  test('has only one dishType selected at a time', () => {
    const mockProps = createMockProps('breakfast');
    const { getAllByRole } = render(<DishForm {...mockProps} />);
    const buttons = getAllByRole('radio') as HTMLInputElement[];
    buttons.forEach((button) => {
      if (button.value !== mockProps.dishType) {
        expect(button.checked).toBe(false);
      } else {
        expect(button.checked).toBe(true);
      }
    });
  });

  test('has each label correctly associated with its input', () => {
    const mockProps = createMockProps('breakfast');
    const { getByLabelText } = render(<DishForm {...mockProps} />);
    const labels = ['Breakfast', 'Lunch', 'Dinner', 'Dessert'];
    labels.forEach((label) => {
      const button = getByLabelText(label) as HTMLInputElement;
      expect(button).toBeTruthy();
    });
  });
});
