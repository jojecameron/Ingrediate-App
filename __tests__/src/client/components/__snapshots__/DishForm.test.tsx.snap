// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DishForm renders a basic snapshot 1`] = `
<div>
  <div
    class="dish-form"
  >
    <form
      id="checkboxes"
    >
      <div
        class="radio-label"
      >
        <label
          for="breakfast"
        >
          Breakfast
        </label>
        <input
          checked=""
          class="checkbox"
          id="breakfast"
          name="breakfast"
          type="radio"
          value="breakfast"
        />
      </div>
      <div
        class="radio-label"
      >
        <label
          for="lunch"
        >
          Lunch
        </label>
        <input
          class="checkbox"
          id="lunch"
          name="lunch"
          type="radio"
          value="lunch"
        />
      </div>
      <div
        class="radio-label"
      >
        <label
          for="dinner"
        >
          Dinner
        </label>
        <input
          class="checkbox"
          id="dinner"
          name="dinner"
          type="radio"
          value="dinner"
        />
      </div>
      <div
        class="radio-label"
      >
        <label
          for="dessert"
        >
          Dessert
        </label>
        <input
          class="checkbox"
          id="dessert"
          name="dessert"
          type="radio"
          value="dessert"
        />
      </div>
    </form>
  </div>
</div>
`;

exports[`DishForm renders a snapshot for dishType: breakfast 1`] = `
<div>
  <div
    class="dish-form"
  >
    <form
      id="checkboxes"
    >
      <div
        class="radio-label"
      >
        <label
          for="breakfast"
        >
          Breakfast
        </label>
        <input
          checked=""
          class="checkbox"
          id="breakfast"
          name="breakfast"
          type="radio"
          value="breakfast"
        />
      </div>
      <div
        class="radio-label"
      >
        <label
          for="lunch"
        >
          Lunch
        </label>
        <input
          class="checkbox"
          id="lunch"
          name="lunch"
          type="radio"
          value="lunch"
        />
      </div>
      <div
        class="radio-label"
      >
        <label
          for="dinner"
        >
          Dinner
        </label>
        <input
          class="checkbox"
          id="dinner"
          name="dinner"
          type="radio"
          value="dinner"
        />
      </div>
      <div
        class="radio-label"
      >
        <label
          for="dessert"
        >
          Dessert
        </label>
        <input
          class="checkbox"
          id="dessert"
          name="dessert"
          type="radio"
          value="dessert"
        />
      </div>
    </form>
  </div>
</div>
`;

exports[`DishForm renders a snapshot for dishType: dessert 1`] = `
<div>
  <div
    class="dish-form"
  >
    <form
      id="checkboxes"
    >
      <div
        class="radio-label"
      >
        <label
          for="breakfast"
        >
          Breakfast
        </label>
        <input
          class="checkbox"
          id="breakfast"
          name="breakfast"
          type="radio"
          value="breakfast"
        />
      </div>
      <div
        class="radio-label"
      >
        <label
          for="lunch"
        >
          Lunch
        </label>
        <input
          class="checkbox"
          id="lunch"
          name="lunch"
          type="radio"
          value="lunch"
        />
      </div>
      <div
        class="radio-label"
      >
        <label
          for="dinner"
        >
          Dinner
        </label>
        <input
          class="checkbox"
          id="dinner"
          name="dinner"
          type="radio"
          value="dinner"
        />
      </div>
      <div
        class="radio-label"
      >
        <label
          for="dessert"
        >
          Dessert
        </label>
        <input
          checked=""
          class="checkbox"
          id="dessert"
          name="dessert"
          type="radio"
          value="dessert"
        />
      </div>
    </form>
  </div>
</div>
`;

exports[`DishForm renders a snapshot for dishType: dinner 1`] = `
<div>
  <div
    class="dish-form"
  >
    <form
      id="checkboxes"
    >
      <div
        class="radio-label"
      >
        <label
          for="breakfast"
        >
          Breakfast
        </label>
        <input
          class="checkbox"
          id="breakfast"
          name="breakfast"
          type="radio"
          value="breakfast"
        />
      </div>
      <div
        class="radio-label"
      >
        <label
          for="lunch"
        >
          Lunch
        </label>
        <input
          class="checkbox"
          id="lunch"
          name="lunch"
          type="radio"
          value="lunch"
        />
      </div>
      <div
        class="radio-label"
      >
        <label
          for="dinner"
        >
          Dinner
        </label>
        <input
          checked=""
          class="checkbox"
          id="dinner"
          name="dinner"
          type="radio"
          value="dinner"
        />
      </div>
      <div
        class="radio-label"
      >
        <label
          for="dessert"
        >
          Dessert
        </label>
        <input
          class="checkbox"
          id="dessert"
          name="dessert"
          type="radio"
          value="dessert"
        />
      </div>
    </form>
  </div>
</div>
`;

exports[`DishForm renders a snapshot for dishType: lunch 1`] = `
<div>
  <div
    class="dish-form"
  >
    <form
      id="checkboxes"
    >
      <div
        class="radio-label"
      >
        <label
          for="breakfast"
        >
          Breakfast
        </label>
        <input
          class="checkbox"
          id="breakfast"
          name="breakfast"
          type="radio"
          value="breakfast"
        />
      </div>
      <div
        class="radio-label"
      >
        <label
          for="lunch"
        >
          Lunch
        </label>
        <input
          checked=""
          class="checkbox"
          id="lunch"
          name="lunch"
          type="radio"
          value="lunch"
        />
      </div>
      <div
        class="radio-label"
      >
        <label
          for="dinner"
        >
          Dinner
        </label>
        <input
          class="checkbox"
          id="dinner"
          name="dinner"
          type="radio"
          value="dinner"
        />
      </div>
      <div
        class="radio-label"
      >
        <label
          for="dessert"
        >
          Dessert
        </label>
        <input
          class="checkbox"
          id="dessert"
          name="dessert"
          type="radio"
          value="dessert"
        />
      </div>
    </form>
  </div>
</div>
`;
